app/services/agenda-service.js

```diff
@@ -182,6 +184,34 @@ export default class AgendaService extends Service {
     const agendaitem = await this.store.findRecord('agendaitem', json.data.id);
     await subcase.hasMany('agendaActivities').reload();
     await subcase.hasMany('submissionActivities').reload();
+    if (json.data.didReorder) {
+      const meeting = await this.store.queryOne('meeting', {
+        'filter[agendas][agendaitems][:id:]': json.data.id,
+      });
+      if (meeting?.id && agendaitem?.number) {
+        const reports = await this.store.queryAll('report', {
+          'filter[:has-no:next-piece]': true,
+          'filter[:has:piece-parts]': true,
+          'filter[decision-activity][treatment][agendaitems][agenda][created-for][:id:]': meeting.id,
+          'filter[decision-activity][treatment][agendaitems][type][:uri:]': CONSTANTS.AGENDA_ITEM_TYPES.NOTA, // announcements are not reordered
+          'filter[decision-activity][treatment][agendaitems][:gt:number]': agendaitem.number, // any report with a lower number should be ok
+        });
+        if (reports?.length) {
+          await Promise.all(reports.map(async (report) => {
+            const agendaitem = await this.store.queryOne('agendaitem', {
+              'filter[:has-no:next-version]': true,
+              'filter[treatment][decision-activity][report][:id:]': report.id,
+            });
+            const documentContainer = await report.documentContainer;
+            const pieces = await documentContainer.pieces;
+            report.name = await generateReportName(agendaitem, meeting, pieces.length);
+            await report.belongsTo('file').reload();
+            await report.save();
```

Comment:
this is a duplication, not sure if we should turn that into a util or service call too.